File: java/SR-RBD/src/ddb/db/DBException.java
/**
 * 
 */
package ddb.db;

import java.io.IOException;
import java.sql.SQLException;

import ddb.BinarySerializable;
import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;

/** 
 * <!-- begin-UML-doc -->
 * Wyjatek rzucany, gdy wystapi blad w bazie danych.
 * <!-- end-UML-doc -->
 * @author User
 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
 */
public class DBException extends Exception implements BinarySerializable
{
	private static final long serialVersionUID = 1L;

	public DBException(SQLException e)
	{
		errorMessage = e.getMessage();
		errorCode = Integer.toString(e.getErrorCode());
	}
	public DBException(DataInputStream s) throws IOException {
		fromBinary(s);
	}

	private String errorMessage;
	private String errorCode;

	public String getErrorMessage()
	{
		return errorMessage;
	}

	public String getErrorCode()
	{
		return errorCode;
	}
	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		errorMessage = s.readString();
		errorCode = s.readString();
		
	}
	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		s.writeString(errorMessage);
		s.writeString(errorCode);
	}
}

File: trunk/C#/RBD/RBD/DB/DBException.cs
// + TODO check

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Communication;

namespace RBD.DB
{
    public class DBException : System.Exception, BinarySerializable
    {
        String vErrorMessage;
        public String ErrorMessage { get { return vErrorMessage; } }
        String vErrorCode;
        public String ErrorCode { get { return vErrorCode; } }

        public DBException(System.Data.SqlClient.SqlException e)
        {
            vErrorMessage = e.Message;
            vErrorCode = e.ErrorCode.ToString();
        }

        public DBException(String msg)
        {
            vErrorMessage = msg;
            vErrorCode = "Parsing 1";
        }

        public DBException(DataInputStream dis) //throws IOException 
        {
            FromBinary(dis);
        }

        public void FromBinary(DataInputStream dis) //throws IOException 
        {
            vErrorMessage = dis.ReadString();
            vErrorCode = dis.ReadString();
        }

        public void ToBinary(DataOutputStream dos) //throws IOException 
        {
            dos.WriteString(ErrorMessage);
            dos.WriteString(ErrorCode);
        }
    }
}
